webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/parameters/GlobalParameter.scala

```diff
@@ -37,32 +37,36 @@
 package com.normation.rudder.domain.parameters
 import java.util.regex.Pattern
 
+import com.normation.rudder.domain.nodes.GenericPropertyUtils
+import net.liftweb.json._
+
 final case class ParameterName(value:String) extends AnyVal
 
 object ParameterName {
   val patternName = Pattern.compile("[a-zA-Z0-9_]+");
 }
 
-/**
- * A Parameter is an object that has a name and a value, to store and reuse
- * values at different places within Rudder
- */
-sealed trait Parameter {
-  def name        : ParameterName
-  def value       : String
-  def description : String
-  def overridable : Boolean
-}
-
 /**
  * A Global Parameter is a parameter globally defined, that may be overriden
  */
 final case class GlobalParameter(
-    override val name       : ParameterName
-  , override val value      : String
-  , override val description: String
-  , override val overridable: Boolean
-) extends Parameter {
+    name       : ParameterName
+  , value      : JValue
+  , description: String
+)
+
+object GlobalParameter {
 
+  /**
+   * A builder with the logic to handle the value part.
+   *
+   * For compatibity reason, we want to be able to process
+   * empty (JNothing) and primitive types, especially string, specificaly as
+   * a JString *but* a string representing an actual JSON should be
+   * used as json.
+   */
+  def apply(name: String, value: String, description: String =""): GlobalParameter = {
```

Comment:
Don(t you think we would want to rename GlobalParameter to GlobalProperty so that we can relate them more easily ?

Reply:
It may make a lot of sense, but it seems to be a diruptive change for a 6.1: breaking API, doc (perhaps internal things). 
I would prefer to keep that kind of things for 7.0 cc @amousset for feedback